### PR TITLE
Add spot price support

### PR DESCRIPTION
**Pull request by jfeala**
Expose existing flag in spark-ec2 script

### CODE REVIEW

**Comment by laserson** [comment]
Thanks, @jfeala, this will be useful!

File: eggo/spark_ec2.py [hunk 1]
```diff
@@ -22,10 +22,12 @@
 def provision():
     provision_cmd = ('{spark_home}/ec2/spark-ec2 -k {ec2_key_pair} '
                      '-i {ec2_private_key_file} -s {slaves} -t {type_} '
-                     '-r {region} {zone_arg} '
+                     '-r {region} {zone_arg} {spot_price_arg} '
                      '--copy-aws-credentials launch {stack_name}')
     az = eggo_config.get('spark_ec2', 'availability_zone')
     zone_arg = '--zone {0}'.format(az) if az != '' else ''
+    spot_price = eggo_config.get('spark_ec2', 'spot_price')
+    spot_price_arg = '--spot-price {}'.format(spot_price) if spot_price != '' else ''
```
Comment (laserson) [inline eggo/spark_ec2.py]: Make python 2.6 compatible by changing to `{0}`.

**Comment by laserson** [comment]
Also, I need to document this better, but would you mind making the same change to all the `.cfg` files in the repo?  I am trying to ensure they stay consistent at the moment, since the config is still somewhat in flux.  You can run `test/jenkins/run-validate-configs.sh` to do it for you.  (note: that script will create a virtualenv).

**Comment by jfeala** [comment]
sure, you just mean `eggo-director.cfg`, right?

**Comment by laserson** [comment]
```
./conf/eggo/eggo-director.cfg
./conf/eggo/eggo.cfg
./test/jenkins/conf/eggo.jenkins.local.cfg
./test/jenkins/conf/eggo.jenkins.spark_ec2.cfg
```

**Comment by laserson** [comment]
Manually squashed and merged in 883d20d.  Thanks @jfeala!

**Pull request closed without merging**